src/index.ml

```diff
@@ -422,9 +422,22 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
         IO.set_generation t.log generation;
         t.generation <- generation
 
-  let force_merge t key value =
+  let get_witness t =
+    let exception Found of entry in
+    try
+      Tbl.iter (fun _ entry -> raise (Found entry)) t.log_mem;
+      match t.index with
+      | None -> None
+      | Some index ->
+          let buf = Bytes.create entry_size in
+          let n = IO.read index.io ~off:0L buf in
+          assert (n = entry_size);
+          Some (decode_entry buf 0)
+    with Found e -> Some e
```

Comment:
You could `try` only the `Tbl.iter ` line with `match Tbl.iter .. with exception Found -> ... | () -> ...` to avoid having unrelated code in the `try` block.